src/core/mac/mac.cpp

```diff
@@ -1154,6 +1156,11 @@ ThreadError Mac::ProcessReceiveSecurity(Frame &aFrame, const Address &aSrcAddr,
             mKeyManager.SetCurrentKeySequence(keySequence);
         }
     }
+    else if (keyIdMode == Frame::kKeyIdMode2)
+    {
+        // reset frame counter
+        mKeyIdMode2FrameCounter = 0;
```

Comment:
According to spec 7.2.1.3, The Frame Counter content of the Key ID Mode 2 Device Descriptor MUST be reset to 0x00000000 after each successful processing of an incoming Key Mode = 2, Key Source = 0xffffffff, and Key Index = 0xff message. So here reset the frame counter as well. Please correct me if my understanding is wrong.